packages/edit-site/src/components/page-patterns/use-patterns.js

```diff
@@ -298,11 +274,7 @@ export const usePatterns = (
 	return useSelect(
 		( select ) => {
 			if ( postType === TEMPLATE_PART_POST_TYPE ) {
-				return selectTemplatePartsAsPatterns(
-					select,
-					categoryId,
-					search
-				);
+				return selectTemplateParts( select, categoryId, search );
```

Comment:
I've only found two uses of the `usePatterns` hook:

- One is in the page-patterns screen, also adapted in this PR.
- The other is in the [sidebar](https://github.com/WordPress/gutenberg/blob/c66631de63dc4cc9ce279ec5fd8e873004a1ed0e/packages/edit-site/src/components/sidebar-navigation-screen-patterns/use-pattern-categories.js#L28), that doesn't need any adaption as it's only used to retrieve data from patterns but not from template parts post types (what is being modified in this PR).

Reply:
I presume that, back in the days, the `usePatterns` hook was created as a way to avoid if conditions in the page (rules of hooks) and also to provide a stable structure to consumers (sidebar, page) independently from where the data came from. It seems to me the consumers still do a lot of data wrangling depending on the type, so not sure how effective this has been for the second goal.

It'd be good to rename it to `usePatternsOrParts` of something similar for clarity at some point: not a requirement for this PR.